proxy/proxy.go

```diff
@@ -142,9 +142,7 @@ func (d *Proxy) Director(r *http.Request) {
 	}
 	*r = *r.WithContext(context.WithValue(r.Context(), ContextKeySession, s))
 
-	for h := range s.Header {
-		r.Header.Set(h, s.Header.Get(h))
-	}
+	CopyHeaders(s.Header, r)
```

Comment:
Wouldn't it be easier to just copy the key/value pairs?

```suggestion
	for k, v := range s.Header {
 		r.Header[k] = v
 	}
```

Reply:
That's would be easier, but I wanted to have canonical header names when oathkeeper returns it to the upstream. That's why I try to use `Get`/`Set` first.

Reply:
Anyway I implemented a cleaner solution